specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json

```diff
@@ -1047,47 +1042,43 @@
           }
         },
         "x-ms-examples": {
-          "Get details of the user provided function apps registered with a static site build.": {
-            "$ref": "./examples/GetUserProvidedFunctionAppsForStaticSiteBuild.json"
+          "Delete a database connection from a static site build.": {
+            "$ref": "./examples/DeleteStaticSiteBuildDatabaseConnection.json"
           }
-        },
-        "x-ms-pageable": {
-          "nextLinkName": "nextLink"
         }
-      }
-    },
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/staticSites/{name}/builds/{environmentName}/userProvidedFunctionApps/{functionAppName}": {
-      "get": {
+      },
+      "patch": {
         "tags": [
           "StaticSites"
         ],
-        "summary": "Gets the details of the user provided function app registered with a static site build",
-        "description": "Description for Gets the details of the user provided function app registered with a static site build",
-        "operationId": "StaticSites_GetUserProvidedFunctionAppForStaticSiteBuild",
+        "summary": "Create or update a database connection for a static site build",
+        "description": "Description for Create or update a database connection for a static site build",
+        "operationId": "StaticSites_UpdateBuildDatabaseConnection",
         "parameters": [
           {
             "$ref": "#/parameters/resourceGroupNameParameter"
           },
           {
             "name": "name",
             "in": "path",
-            "description": "Name of the static site.",
+            "description": "Name of the static site",
             "required": true,
             "type": "string"
           },
           {
-            "name": "environmentName",
-            "in": "path",
-            "description": "The stage site identifier.",
-            "required": true,
-            "type": "string"
+            "$ref": "#/parameters/environmentNameParameter"
           },
           {
-            "name": "functionAppName",
-            "in": "path",
-            "description": "Name of the function app registered with the static site build.",
+            "$ref": "#/parameters/databaseConnectionNameParameter"
+          },
+          {
+            "name": "databaseConnectionRequestEnvelope",
+            "in": "body",
+            "description": "A JSON representation of the database connection request properties",
             "required": true,
-            "type": "string"
+            "schema": {
+              "$ref": "#/definitions/DatabaseConnectionPatchRequest"
```

Comment:
>DatabaseConnectionPatchRequest [](http://example.com/codeflow?start=37&length=30)

Can you just make the schema of the patch request 'DatabaseConnectionPatchRequest'? And add x-ms-mutability: "create,read" notations on any properties which aren't upddatable? OR does it really need to be different?

Reply:
I think I really do need a separate schema because of this discussion https://github.com/Azure/azure-rest-api-specs/pull/22159#discussion_r1080649625. Otherwise the ProxyOnlyResource inheritance causes issues for the PATCH requests.